File: src/main/java/com/github/graqr/threshr/model/redsky/products/Options.java
package com.github.graqr.threshr.model.redsky.products;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable
public record Options(
        @JsonProperty("display_name")
        String displayName,
        @JsonProperty("canonical_name")
        String canonicalName,
        String id,
        @Nullable String source,
        @JsonProperty("image_url")
        @Nullable String imageUrl,
        String url,
        String value
) {
}

File: src/main/kotlin/com/graqr/threshr/model/redsky/product/Options.kt
package com.graqr.threshr.model.redsky.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.core.annotation.Nullable
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class Options(
    @param:JsonProperty("display_name") val displayName: String,
    @param:JsonProperty("canonical_name") val canonicalName: String,
    val id: String,
    @param:Nullable val source: String,
    @param:JsonProperty("image_url") @param:Nullable val imageUrl: String,
    val url: String,
    val value: String
) {}
